Changes in class_create() and DEFINE_SEMAPHORE() introduced in kernel 6.4
by the following commits:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=1aaba11da9aa7d7d6b52a74d45b31cac118295a1
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=48380368dec14859723b9e3fbd43e042638d9a76

---
 quickassist/qat/drivers/crypto/qat/qat_common/adf_ctl_drv.c   | 2 +-
 .../qat/drivers/crypto/qat/qat_common/adf_dev_processes.c     | 4 ++--
 quickassist/qat/drivers/crypto/qat/qat_common/qat_dbg_drv.c   | 2 +-
 .../utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c    | 2 +-
 .../utilities/osal/src/linux/kernel_space/OsalThread.c        | 4 ++--
 .../utilities/osal/src/linux/kernel_space/OsalUsrKrlProxy.c   | 3 +--
 6 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/quickassist/qat/drivers/crypto/qat/qat_common/adf_ctl_drv.c b/quickassist/qat/drivers/crypto/qat/qat_common/adf_ctl_drv.c
index 10624f8..8e40e91 100644
--- a/quickassist/qat/drivers/crypto/qat/qat_common/adf_ctl_drv.c
+++ b/quickassist/qat/drivers/crypto/qat/qat_common/adf_ctl_drv.c
@@ -161,7 +161,7 @@ static int adf_chr_drv_create(void)
 		return -EFAULT;
 	}
 
-	adf_ctl_drv.drv_class = class_create(THIS_MODULE, DEVICE_NAME);
+	adf_ctl_drv.drv_class = class_create(DEVICE_NAME);
 	if (IS_ERR(adf_ctl_drv.drv_class)) {
 		pr_err("QAT: class_create failed for adf_ctl\n");
 		goto err_chrdev_unreg;
diff --git a/quickassist/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c b/quickassist/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
index 08e5bea..ad41dbb 100644
--- a/quickassist/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
+++ b/quickassist/qat/drivers/crypto/qat/qat_common/adf_dev_processes.c
@@ -60,7 +60,7 @@ static struct adf_chr_drv_info adf_processes_drv_info = {
 };
 
 static LIST_HEAD(processes_list);
-static DEFINE_SEMAPHORE(processes_list_sema);
+static DEFINE_SEMAPHORE(processes_list_sema, 1);
 
 static void adf_chr_drv_destroy(void)
 {
@@ -83,7 +83,7 @@ static int adf_chr_drv_create(void)
 	}
 
 	adf_processes_drv_info.drv_class =
-		class_create(THIS_MODULE, ADF_DEV_PROCESSES_NAME);
+		class_create(ADF_DEV_PROCESSES_NAME);
 	if (IS_ERR(adf_processes_drv_info.drv_class)) {
 		pr_err("QAT: class_create failed for adf_ctl\n");
 		goto err_chrdev_unreg;
diff --git a/quickassist/qat/drivers/crypto/qat/qat_common/qat_dbg_drv.c b/quickassist/qat/drivers/crypto/qat/qat_common/qat_dbg_drv.c
index 1868f36..d90542d 100644
--- a/quickassist/qat/drivers/crypto/qat/qat_common/qat_dbg_drv.c
+++ b/quickassist/qat/drivers/crypto/qat/qat_common/qat_dbg_drv.c
@@ -405,7 +405,7 @@ static int qat_dbg_chr_drv_create(void)
 		return -EFAULT;
 	}
 
-	qatd_drv.drv_class = class_create(THIS_MODULE, QATD_DEVICE_NAME);
+	qatd_drv.drv_class = class_create(QATD_DEVICE_NAME);
 	if (IS_ERR(qatd_drv.drv_class)) {
 		pr_err("QAT: class_create failed for qat_dbg\n");
 		goto err_chrdev_unreg;
diff --git a/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c b/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
index 0cf9733..c2bef72 100644
--- a/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
+++ b/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
@@ -1197,7 +1197,7 @@ chr_drv_create_class(chr_drv_info_t* drv_info)
                      "chr_drv_create_class(): Invalid parameter value ",
                      -EINVAL);
 
-    drv_info->drv_class = class_create(THIS_MODULE, drv_info->name);
+    drv_info->drv_class = class_create(drv_info->name);
     if (IS_ERR(drv_info->drv_class))
     {
         mm_err("%s:%d class_create failed\n",__func__,__LINE__);
diff --git a/quickassist/utilities/osal/src/linux/kernel_space/OsalThread.c b/quickassist/utilities/osal/src/linux/kernel_space/OsalThread.c
index c7b5839..b2c1927 100644
--- a/quickassist/utilities/osal/src/linux/kernel_space/OsalThread.c
+++ b/quickassist/utilities/osal/src/linux/kernel_space/OsalThread.c
@@ -78,8 +78,8 @@
 DECLARE_MUTEX (osalThreadMutex);
 DECLARE_MUTEX (osalThreadStopMutex);
 #else
-DEFINE_SEMAPHORE (osalThreadMutex);
-DEFINE_SEMAPHORE (osalThreadStopMutex);
+DEFINE_SEMAPHORE (osalThreadMutex, 1);
+DEFINE_SEMAPHORE (osalThreadStopMutex, 1);
 #endif
 
 OSAL_PUBLIC OSAL_STATUS
diff --git a/quickassist/utilities/osal/src/linux/kernel_space/OsalUsrKrlProxy.c b/quickassist/utilities/osal/src/linux/kernel_space/OsalUsrKrlProxy.c
index fe81da1..8886220 100644
--- a/quickassist/utilities/osal/src/linux/kernel_space/OsalUsrKrlProxy.c
+++ b/quickassist/utilities/osal/src/linux/kernel_space/OsalUsrKrlProxy.c
@@ -621,8 +621,7 @@ chr_drv_create_class(chr_drv_info_t* drv_info, char* path)
 
     }
 
-    drv_info->drv_class = class_create(THIS_MODULE,
-                       (path) ? name : drv_info->name);
+    drv_info->drv_class = class_create((path) ? name : drv_info->name);
     if (IS_ERR(drv_info->drv_class)) {
         osalLog(
             OSAL_LOG_LVL_ERROR, OSAL_LOG_DEV_STDOUT, "class_create failed\n");
-- 
